### DISCUSSION TITLE
KeyError "Unable to load frame" causing SLEAP to crash when predicting

### DISCUSSION THREAD

**Discussion by lencriv** [talmolab/sleap#767] [Help!] [answered]
# Problem description

When predicting over an entire video the program crashes

<details><summary>Logs</summary>

```
KeyError: "Unable to load frame 17 from MediaVideo(filename='C:/Users/LEncR/Desktop/FM_H&T/FM_H_T/mouse_behavior/day 1 vs DAC/mr8/mr8_105_v0-1_e366b7a7-d2f5-11ec-9668-38fc98ee8dd7/mr4_27_v0-1_12ede32a-8ab7-11ec-9ca2-38fc98ee8dd7_living_zone_vid.mp4', grayscale=True, bgr=True, dataset='', input_format='channels_last')."
```

</details>

## Screenshots

![image](https://user-images.githubusercontent.com/47608464/170735885-74e3ab50-bded-47b5-9d16-c1ef9e5249f9.png)

**Comment by roomrys** [comment, marked as answer]
Hi @lencriv,

We have seen a few problems like this occur when the video is not [reliably seekable](http://blog.loopbio.com/video-io-1-introduction.html). Try [reencoding your video](https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format) and let us know how that works.

---
#### Update: If reencoding doesn't work, and results in `moov atom not found`
As I was helping a user yesterday, we found that they had just copied the contents of a zipped file over without unzipping which resulted in first the `KeyError`, and then, when attempting to reencode, a `moov atom not found` error. After retrying but with **unzipping first**, we were able to get everything running smoothly.

---

Thanks,
Liezl

P.S. Also linking similar problems here for reference (namely Unable to load frame #366, KeyError when trying to run inference #630, Encoding Multiple Videos using Batch File #719, Prediction all frames in a video and processed to 99% and error #742).

**Comment by lencriv** [comment]
Hi @roomrys 
this worked! thanks :)

**Comment by roomrys** [comment]
Great! I am converting this to a discussion in case others might find it helpful.

**Comment by jmdelahanty** [comment]
Related thing I've encountered with attempting to just copy contents from .mkv into .mp4 with ffmpeg like this:

`ffmpeg -i output_file.mkv -c copy final_output.mp4`

ffprobe gave me this for the original .mkv file:

```
ffprobe version 9c33b2f Copyright (c) 2007-2021 the FFmpeg developers
  built with gcc 9.3.0 (crosstool-NG 1.24.0.133_b0863d8_dirty)
  configuration: --prefix=/home/conda/feedstock_root/build_artifacts/ffmpeg_1627813612080/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac --cc=/home/conda/feedstock_root/build_artifacts/ffmpeg_1627813612080/_build_env/bin/x86_64-conda-linux-gnu-cc --disable-doc --disable-openssl --enable-avresample --enable-gnutls --enable-gpl --enable-hardcoded-tables --enable-libfreetype --enable-libopenh264 --enable-libx264 --enable-pic --enable-pthreads --enable-shared --enable-static --enable-version3 --enable-zlib --enable-libmp3lame --pkg-config=/home/conda/feedstock_root/build_artifacts/ffmpeg_1627813612080/_build_env/bin/pkg-config
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
Input #0, matroska,webm, from '../raw/20221226_CSE036_CSE037_empty.mkv':
  Metadata:
    ENCODER         : Lavf58.29.100
  Duration: 00:06:35.70, start: 0.000000, bitrate: 2664 kb/s
    Stream #0:0: Video: h264 (High), yuv420p(tv, bt709, progressive), 1920x1080, 30 fps, 30 tbr, 1k tbn, 60 tbc (default)
    Metadata:
      DURATION        : 00:06:35.700000000
    Stream #0:1: Audio: aac (LC), 48000 Hz, stereo, fltp (default)
    Metadata:
      title           : simple_aac
      DURATION        : 00:06:35.584000000
 ```
 
 So even though the video was encoded with 264, getting it into the .mp4 container still gave me an issue in multiple videos collected allegedly with VeeCool Video Capture (please for the love of god use anything else for video acquisition).
